common_knowledge/ARCHITECTURE.md

```diff
@@ -0,0 +1,505 @@
+
+### 5.1 Environment Variables
+
+- Managed via `.env` files and documented in [Environment-Variables.md](Environment-Variables.md).
+
+### 5.2 App Initialization Flow
+
+- See [App-Initialization-Flow.md](App-Initialization-Flow.md) for details on bootstrapping and dependency wiring.
+
+---
+
+## 6. Event-Driven Architecture
+
+### 6.1 RabbitMQ Usage
+
+- Used for asynchronous, event-driven communication between services and workers.
+
+### 6.2 Outbox Pattern
+
+- Ensures reliable event delivery and recovery from failures.
+
+### 6.3 Idempotency & Retry Strategies
+
+- Policies for handling duplicate or failed events. See [RabbitMQ.md](RabbitMQ.md).
+
+---
+
+## 7. Security & Permissions
+
+### 7.1 Authentication
+
+- OAuth, wallet, and session management.
+
+### 7.2 Authorization
+
+- Tiered access, roles, and middleware.
+
+### 7.3 Secrets Management
+
+- Secure handling of sensitive configuration.
+
+---
+
+## 8. Deployment & Operations
+
+### 8.1 CI/CD
+
+- Build, test, and deployment pipelines.
+
+### 8.2 Monitoring & Logging
+
+- Datadog, Rollbar, and health checks.
+
+### 8.3 Scaling & Fault Tolerance
+
+- Strategies for handling load and failures.
+
+---
+
+## 9. Extensibility & Customization
+
+### 9.1 Adding and Running Jobs/Workers
+
+- **Jobs** such as `graphileWorker`, `knock`, `twitterWorker`, `evmChainEvents`, `commonwealthConsumer`, `messageRelayer`, and `discordBot` are located in `packages/commonwealth/server/workers/`.
+- In production, these are typically run as separate processes or containers.
+- **DEV_MODULITH**: In development mode, you can set the `DEV_MODULITH` environment variable to `true` to run all jobs/workers in the same server process as the API. This simplifies local development and debugging. Example:
+
+  ```sh
+  DEV_MODULITH=true pnpm run start
+  ```
+
+- This will start the API server and all background jobs in a single process, reducing the need for multiple terminals or containers.
+
+### 9.2 Adding New Domain Logic
+
+- **Aggregates**: Create a new folder under `libs/model/src/aggregates/` for your domain entity. Implement commands, queries, and event handlers as needed.
+- **Commands**: Add a `*.command.ts` file exporting a factory function that returns a `Command<Schema>`. Use Zod schemas for input/output validation.
+- **Queries**: Add a `*.query.ts` file for read operations, following the same pattern as commands.
+- **Policies**: If your logic is event-driven or cross-cutting, add a new policy in `libs/model/src/policies/` or extend an existing one. Policies listen to events and can trigger commands or side effects.
+- **Export** your new logic in the relevant `index.ts` files for automatic discovery and registration.
+- **Testing**: Add tests in `libs/model/test/` to cover your new logic.
+- **Documentation**: Update or add documentation in `common_knowledge/` as needed.
+
+---
+
+## 10. Diagrams
+
+### 10.1 Logical Architecture Diagram
+
+```mermaid
+graph TD
```

Comment:
These diagrams don't seem right. Most workers don't use/interact with the API server and none of them interact directly with RabbitMQ (except for MessageRelayer of course).

Reply:
AI again, but let's talk in future OHs